cmake/CompilerWarnings.cmake

```diff
@@ -0,0 +1,90 @@
+function(set_enabled_compiler_warnings target_name)
+    option(ENABLE_WARNINGS "Enable compiler warnings" OFF)
+    option(WARNINGS_AS_ERRORS "Treat warnings as erros" OFF)
+
+    if (ENABLE_WARNINGS)
+        set(MSVC_WARNINGS
```

Comment:
Even though this is not going to be used on non-Windows, it’s a bit silly to declare variables when not needed. Not very important, but still a bit iffy.